compiler+runtime/src/cpp/jank/jit/processor.cpp

```diff
@@ -16,6 +16,18 @@
 
 namespace jank::jit
 {
+
+  inline native_persistent_string shared_lib_name(native_persistent_string const &lib)
+#if defined(__APPLE__)
+  {
+    return fmt::format("{}.dylib", lib);
+  }
+#elif defined(__linux__)
+  {
+    return fmt::format("lib{}.so", lib);
```

Comment:
There's an interesting thing with shared object versioning and I'm not sure if it's going to bite us. There can be a `libfoo.so.1`, for example, and `libfoo.so.2`. Will you do some digging into this, to see if it's anything we need to support? I'll also do the same.

Reply:
https://unix.stackexchange.com/a/476 implies that `-lfoo` with gcc/clang can't specify the version number either. So we can rely on the `libfoo.so.2` form or the absolute path form for that. We just need to not assume that it ends in `.so` anywhere, since it may not!

Reply:
done, now we check if `libfoo.so` exists if `-lfoo` was provided as an argument to the cli.

In addition to that, we also check for `dynamic-lib-path/foo`. This handles the case where the library name is given as `-llibfoo.so.1`